### ISSUE TITLE
Add Request Interceptor REST API

### ISSUE THREAD

**Issue by pavel-paulau** [webmetrics/browsermob-proxy#6]
Please implement such important feature!

**Comment by lightbody** [comment]
Yes, I want to get this implemented soon. It will basically be an API where you can send the proxy a snippet of JavaScript that we will execute for every HTTP request (via a callback).

**Comment by pavel-paulau** [comment]
That sounds great, I look forward to it.
